src/dataframe/dataframe.jl

```diff
@@ -368,7 +380,7 @@ end
 function Base.setindex!(df::DataFrame,
                 v::Any,
                 col_ind::ColumnIndex)
-    insert_single_column!(df, upgrade_scalar(df, v), col_ind)
+    fill_single_column!(df, v, col_ind)
```

Comment:
But you're right in that the `fill_single_column!` function is just `setindex!`. I suppose it would be easier just to inline this function.